subxt/src/client/online_client.rs

```diff
@@ -61,7 +61,10 @@ impl<T: Config> std::fmt::Debug for OnlineClient<T> {
 }
 
 // The default constructors assume Jsonrpsee.
-#[cfg(feature = "jsonrpsee")]
+#[cfg(any(
+    all(feature = "jsonrpsee-ws", not(target_arch = "wasm32")),
+    all(feature = "jsonrpsee-web", target_arch = "wasm32")
+))]
```

Comment:
Should we have a check at the root lib.rs to make sure that somebody cannot enable both of these features?

What are the pros and cons for also having this `target_arch` check as well as just checking the features? It'd be ncie if we could just check for a feature and not worry about the rest, for simplicity?

Reply:
> Should we have a check at the root lib.rs to make sure that somebody cannot enable both of these jsonrpsee features simultaneously?

Sure

> What are the pros and cons for also having this target_arch check as well as just checking the features? It'd be nice if we could just check for a feature and not worry about the rest, for simplicity

It's my own fault in [jsonrpsee](https://github.com/paritytech/jsonrpsee/blob/master/core/src/client/async_client/mod.rs#L196) i.e, it won't compile with the target_arch

Reply:
Ok that makes sense! Is it worth filing an issue/PR in jsonrpsee to tweak that so we can simplify here?

Reply:
yeah, I can take a look if it's possible to remove those